receiver/githubreceiver/internal/traces/config.go

```diff
@@ -0,0 +1,40 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package traces // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/githubreceiver/internal/traces"
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/multierr"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/webhookeventreceiver"
+)
+
+var errMissingEndpointFromConfig = errors.New("missing receiver server endpoint from config")
+
+// Config defines configuration for GitHub Actions receiver
+type Config struct {
+	WebhookReceiver   webhookeventreceiver.Config `mapstructure:",squash"`             // squash ensures fields are correctly decoded in embedded struct
+	Secret            string                      `mapstructure:"secret"`              // GitHub traces signature. Default is empty
+	CustomServiceName string                      `mapstructure:"custom_service_name"` // custom service name. Default is empty
+	ServiceNamePrefix string                      `mapstructure:"service_name_prefix"` // service name prefix. Default is empty
+	ServiceNameSuffix string                      `mapstructure:"service_name_suffix"` // service name suffix. Default is empty
+}
+
+var _ component.Config = (*Config)(nil)
+
+// Validate checks the receiver configuration is valid
+func (cfg *Config) Validate() error {
```

Comment:
TODO: revisit this after the discussion of using the webhook receiver as part of this or re-implementing similar code.